src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/PowCoinviewRule.cs

```diff
@@ -128,18 +101,18 @@ public virtual void ExecuteBlock(RuleContext context, TaskScheduler taskSchedule
                         var txData = new PrecomputedTransactionData(tx);
                         for (int inputIndex = 0; inputIndex < tx.Inputs.Count; inputIndex++)
```

Comment:
Created issue https://github.com/stratisproject/StratisBitcoinFullNode/issues/1432  for a potential improvement to break this down into sub rules or sub component that run inside the transaction loop but are individually testable/reusable and more SOLID compliant than now.